backend/src/services/activityService.ts

```diff
@@ -498,174 +498,66 @@ export default class ActivityService extends LoggerBase {
     return exists || false
   }
 
-  async createWithMember(data, fireCrowdWebhooks: boolean = true) {
+  async createWithMember(data) {
     const logger = this.options.log
-    const searchSyncService = new SearchSyncService(this.options, SyncMode.ASYNCHRONOUS)
-
-    const errorDetails: any = {}
-
-    const transaction = await SequelizeRepository.createTransaction(this.options)
-    const memberService = new MemberService(this.options)
+    const dataSinkWorkerEmitter = await getDataSinkWorkerEmitter()
 
     try {
       data.member.username = mapUsernameToIdentities(data.member.username, data.platform)
 
-      const platforms = Object.keys(data.member.username)
-      if (platforms.length === 0) {
-        throw new Error('Member must have at least one platform username set!')
-      }
-
       if (!data.username) {
-        data.username = data.member.username[data.platform][0].username
+        data.username = data.member.username[data.platform][0].value
       }
 
       logger.trace(
         { type: data.type, platform: data.platform, username: data.username },
-        'Creating activity with member!',
+        'Processing activity with member!',
       )
 
-      let activityExists = await this._activityExists(data, transaction)
+      const segment = SequelizeRepository.getStrictlySingleActiveSegment(this.options)
 
-      let existingMember = activityExists
-        ? await memberService.findById(activityExists.memberId, true, false)
-        : false
+      data.member.identities = ActivityService.processMemberIdentities(data.member, data.platform)
+      data.isContribution = data.isContribution || false
 
-      if (existingMember) {
-        // let's look just in case for an existing member and if they are different we should log them because they will probably fail to insert
-        const tempExisting = await memberService.memberExists(data.member.username, data.platform)
+      // prepare objectMember for dataSinkWorker
+      if (data.objectMember) {
+        data.objectMember.username = mapUsernameToIdentities(
+          data.objectMember.username,
+          data.platform,
+        )
+        data.objectMemberUsername = data.objectMember.username[data.platform][0].value
```

Comment:
I think it might be better to check if this ( `data.objectMember.username[data.platform][0].value` ) exists and if not throw an error here